flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/common/RMQConnectionConfig.java

```diff
@@ -90,12 +94,16 @@ private RMQConnectionConfig(
             Integer requestedChannelMax,
             Integer requestedFrameMax,
             Integer requestedHeartbeat,
-            Integer prefetchCount) {
+            Integer prefetchCount,
+            Long deliveryTimeout) {
         Preconditions.checkNotNull(host, "host can not be null");
         Preconditions.checkNotNull(port, "port can not be null");
         Preconditions.checkNotNull(virtualHost, "virtualHost can not be null");
         Preconditions.checkNotNull(username, "username can not be null");
         Preconditions.checkNotNull(password, "password can not be null");
+        Preconditions.checkArgument(
+                deliveryTimeout == null || deliveryTimeout >= 0,
+                "deliveryTimeout can not be negative");
```

Comment:
Can we split this into two checks or update the error message to account for null?

Reply:
I think this check should be 
```suggestion
        Preconditions.checkArgument(
                deliveryTimeout != null && deliveryTimeout >= 0,
                "deliveryTimeout can not be negative");
```

Reply:
The check is correct. The value should either be not set at all (so `null` is fine, default will be used here) or set to non-negative number.
It could be refactored a bit, so that the config would `long` instead of nullable `Long` and the default (if not provided) would be set by the builder. But I would have doubts if this is a good idea. Currently the builder is just a helper to build the config instance, with no internal logic.
What's your opinion? Or maybe you have some other ideas how to make it more clear?